pkg/nodeagent/controller/lease/reconciler.go

```diff
@@ -56,6 +56,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		}
 		return nil
 	})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	log.V(1).Info("Heartbeat Lease", "lease", client.ObjectKeyFromObject(lease), "operation", op)
-	return reconcile.Result{RequeueAfter: time.Duration(r.LeaseDurationSeconds) * time.Second / 4}, err
+	return reconcile.Result{RequeueAfter: time.Duration(r.LeaseDurationSeconds) * time.Second / 4}, nil
```

Comment:
Why did you change this?

Reply:
When you return an non-empty `reconcile.Result` and a non-nil error at the same time, controller-runtime writes a warning to the logs. See this commit message https://github.com/gardener/gardener/pull/10014/commits/47d1e33f4245b029c3a19a9f2af78a278f5a80a3